dashboard/app/controllers/courses_controller.rb

```diff
@@ -25,20 +25,16 @@ def index
   end
 
   def show
-    # csp and csd are each "course families", each containing two "course versions",
-    # one with version year 2017 and one with version year 2018. When the url
-    # of a course family is requested, redirect to a specific course version.
+    # csp and csd are each "course families", each containing multiple "course versions".
+    # When the url of a course family is requested, redirect to a specific course version.
     #
     # For now, Hard-code the redirection logic because there are only two course
     # families to worry about. In the future we will want to make this redirect
     # happen based on the data in the DB so it can be configured via levelbuilder.
-    redirect_query_string = request.query_string.empty? ? '' : "?#{request.query_string}"
-    case params[:course_name]
-    when 'csd'
-      redirect_to "/courses/csd-2018#{redirect_query_string}"
-      return
-    when 'csp'
-      redirect_to "/courses/csp-2018#{redirect_query_string}"
+    if ['csd', 'csp'].include?(params[:course_name])
+      redirect_query_string = request.query_string.empty? ? '' : "?#{request.query_string}"
+      course_name = Course.latest_stable_version(params[:course_name]).name
+      redirect_to "/courses/#{course_name + redirect_query_string}"
```

Comment:
While I love this approach, it will add a DB read which is hard to tell if it will be a problem or not. Can we hard-code this to return csd-2019 / csp-2019 for now, and take a follow-up item to add back this generalization and protect it behind a course cache? I think the course cache will solve some problems other places too.

Reply:
ah, yeah, good point. yep, i'll hardcode for now so it can go out with tomorrow's DTP